### DISCUSSION TITLE
Rip subtitles batch from a txt file list

### DISCUSSION THREAD

**Discussion by mdkberry** [SubtitleEdit/subtitleedit#7835] [General]
I have a long list of mkv files in many different folders that I want to rip subtitles from using the Whisper feature. I hoped I could import a list but it doesnt have that option, I tried adding multiples using "file location 1" "file location 2" in the file load field, but you can only add about 3 at the most. Is there a fast way to do this from a txt file would be really cool. Currently I have to do it manually.

**Comment by niksedk** [comment]
Hi @mdkberry,

I think you can just paste (Ctrl+v) in the file list view.

**Reply by mdkberry** [reply]
If you mean you can cut and paste into the "file name" field after you select the "Add" button then you can, but only about 2 or 3 folder paths will fit in there. I used that method to add 3 at a time until I had added all I needed then ran generate. But it really could do with the option to add more in one go.

If there is another way  to cut and paste into the area (see the attached image for where I am talking about), I can't find one. Right clicking doesnt offer the option, and straight up CTRL-V doesnt do anything either. 

![subtitleedit-batch1](https://github.com/SubtitleEdit/subtitleedit/assets/810955/05699d97-d18e-4876-b94a-bd59bdf6ac25)

**Comment by darnn** [comment]
I too am in favor of adding the option to do it via text file. In the meanwhile, the two things you can do are: Display the files in a program like Everything (voidtools.com), select the ones you want and drag and drop them into the above window, or, again, select the ones you want, copy them, then click inside `Input` in the above window, on the empty file list, and press ctrl+v.

**Reply by mdkberry** [reply]
`click inside Input in the above window, on the empty file list, and press ctrl+v`

as mentioned, that doesnt work. but I have just tried dragging an mp4 into it and that did work. 

The problem I have, is that my list of mp4s is spread across many folders. ergo, why I need to upload a txt file based list, not just drag mp4's into it, as I would be hunting round hundreds of folders looking for each one. Neither does Everything work (I have it), because it shows me all mp4's and cannot filter by my txt file list of specific mp4s that need subtitles created.

if there is another way I haven't figured it out yet.

**Comment by darnn** [comment]
If you have a text file, you can take the individual paths from it and pop them into Everything, like this:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/19672444/5360d721-f1b6-43c3-b83b-4534f6c7cc10)


As for pressing ctrl+v not working, I'm not sure what to tell you. Here's me doing it:
![GIF 1-18-2024 12-32-29 AM](https://github.com/SubtitleEdit/subtitleedit/assets/19672444/bad40c5d-1aa0-4aec-a41e-348d17c0e8ad)
As per my post above, I left-click inside Input and press ctrl+v.

**Reply by mdkberry** [reply]
Nice work around! I did not know it would do that. I'll use it next time. Thanks

EDIT: that is weird how it accepts CTRL-V with actually files but not URL paths. Interesting.
